module/data/item/templates/activities.mjs

```diff
@@ -134,4 +204,133 @@ export default class ActivitiesTemplate extends SystemDataModel {
   prepareFinalActivityData(rollData) {
     UsesField.prepareData.call(this, rollData);
   }
+
+  /* -------------------------------------------- */
+  /*  Helpers                                     */
+  /* -------------------------------------------- */
+
+  /**
+   * Retrieve information on available uses for display.
+   * @returns {{value: number, max: number, name: string}}
+   */
+  getUsesData() {
+    return { value: this.uses.value, max: this.uses.max, name: "system.uses.value" };
+  }
+
+  /* -------------------------------------------- */
+  /*  Shims                                       */
+  /* -------------------------------------------- */
+
+  /**
+   * Apply shims for data removed from ActionTemplate & ActivatedEffectTemplate.
+   * @this {ItemDataModel}
+   */
+  static _applyActivityShims() {
+    const shim = (template, property, get) => {
+      if ( property in this ) return;
+      Object.defineProperty(this, property, {
+        get: () => {
+          foundry.utils.logCompatibilityWarning(
+            `The \`${property}\` property on \`${template}\` has been deprecated.`,
+            { since: "DnD5e 4.0", until: "DnD5e 4.4", once: true }
+          );
+          return get();
+        },
+        configurable: true,
+        enumerable: false
+      });
+    };
+    const addShims = (template, shims) => Object.entries(shims).forEach(([key, method]) => shim(template, key, method));
+    const firstActivity = this.activities.contents[0] ?? {};
+
+    addShims("ActionTemplate", {
+      ability: () => firstActivity.ability ?? null,
+      actionType: () => firstActivity.actionType ?? "",
+      attack: () => {
+        const activity = this.activities.getByType("attack")[0] ?? {};
+        return {
+          bonus: activity.attack?.bonus ?? "",
+          flat: activity.attack?.flat ?? false
+        };
+      },
+      chatFlavor: () => firstActivity.description?.chatFlavor ?? "",
+      critical: () => {
+        const activity = this.activities.getByType("attack")[0] || this.activities.getByType("damage")[0];
+        return {
+          threshold: activity?.attack?.critical?.threshold ?? null,
+          damage: activity?.damage?.critical?.bonus ?? ""
+        };
+      },
+      dammage: () => ({ parts: [], versatile: "" }),
+      enchantment: () => this.activities.getByType("enchant")[0],
+      formula: () => this.activities.getByType("utility")[0]?.roll?.formula ?? "",
+      hasAbilityCheck: () => false,
+      hasAttack: () => !!this.activities.getByType("attack").size,
+      hasDamage: () => !!this.activities.find(a => a.damage?.parts?.length),
+      hasSave: () => !!this.activities.getByType("save").size,
+      hasSummoning: () => !!this.activities.getByType("summon").size,
+      isEnchantment: () => !!this.activities.getByType("enchant").size,
+      isHealing: () => !!this.activities.getByType("heal").size,
+      isVersatile: () => this.actionType && this.properties?.has("ver"),
```

Comment:
Some of these seem like they would still be useful. For example, checking if a weapon is versatile, or checking if an Item has an attack rather than calling `item.system.activities.getByType("attack").size` each time. Should they just be moved to `ActivitiesTemplate` rather than throwing a deprecation warning?